docs/source/tutorial/typesfuns.rst

```diff
@@ -33,21 +33,20 @@ into the Idris interactive environment by typing ``idris2 Prims.idr``:
 
 An Idris file consists of an optional module declaration (here
 ``module Prims``) followed by an optional list of imports and a
-collection of declarations and definitions. In this example no imports
-have been specified. However Idris programs can consist of several
-modules and the definitions in each module each have their own
-namespace. This is discussed further in Section
-:ref:`sect-namespaces`. When writing Idris programs both the order in which
+collection of declarations and definitions. In this example, no imports
+have been specified. However, Idris programs can consist of several
+modules, each with its own namespace. This is discussed further in Section
+:ref:`sect-namespaces`. When writing Idris programs, both the order in which
 definitions are given and indentation are significant. Functions and
-data types must be defined before use, incidentally each definition must
-have a type declaration, for example see ``x : Int``, ``foo :
-String``, from the above listing. New declarations must begin at the
+data types must be defined before use. Incidentally, each definition must
+have a type declaration — for example, see ``x : Int`` or ``foo :
```

Comment:
"or" ~> "and", both provide a combined example of type declarations before definition.

Reply:
"or" also implies that both are examples. Moreover, the list is not exhaustive. There's also bar : Char and quux : Bool. "and" would more so imply these are all the examples.